pypeit/images/rawimage.py

```diff
@@ -677,6 +677,7 @@ def process(self, par, bpm=None, scattlight=None, flatimages=None, bias=None, sl
                                               shot_noise=self.par['shot_noise'],
                                               bpm=_bpm.astype(bool))
 
+        pypeitImage.filename = self.filename
```

Comment:
You added `filename` to the argument list for `PypeItImage`.  Can you use that instead of adding this line?  I.e., the lines below are needed because `rawheadlist` and `process_steps` are not keyword arguments for `PypeItImage`.